MantidQt/CustomInterfaces/src/Indirect/AbsorptionCorrections.cpp

```diff
@@ -301,29 +298,44 @@ void AbsorptionCorrections::algorithmComplete(bool error) {
                                          m_uiForm.spCanShift->value()));
     shiftLog->execute();
   }
-  // Plot output
-
-  bool plot = m_uiForm.ckPlot->isChecked();
-  if (plot) {
-    QStringList plotData = {QString::fromStdString(m_pythonExportWsName),
-                            m_uiForm.dsSampleInput->getCurrentDataName()};
-    auto outputFactorsWsName =
-        m_absCorAlgo->getPropertyValue("CorrectionsWorkspace");
-    if (m_uiForm.ckKeepFactors->isChecked()) {
-      QStringList plotCorr = {QString::fromStdString(outputFactorsWsName) +
-                              "_ass"};
-      if (m_uiForm.ckUseCanCorrections->isChecked()) {
-        plotCorr.push_back(QString::fromStdString(outputFactorsWsName) +
-                           "_acc");
-        QString shiftedWs = QString::fromStdString(
-            m_absCorAlgo->getPropertyValue("CanWorkspace"));
-        plotData.push_back(shiftedWs);
-      }
-      plotSpectrum(plotCorr, 0);
+  // Enable plot and save
+  m_uiForm.pbPlot->setEnabled(true);
+  m_uiForm.pbSave->setEnabled(true);
+}
+/**
+ * Handle saving of workspace
+ */
+void AbsorptionCorrections::saveClicked() {
+
+  checkADSForPlotSaveWorkspace(m_pythonExportWsName, false);
+  addSaveWorkspace(QString::fromStdString(m_pythonExportWsName));
+  bool keepCorrectionFactors = m_uiForm.ckKeepFactors->isChecked();
+  checkADSForPlotSaveWorkspace(m_outputFactorsWsName.toStdString(), false);
+  if (keepCorrectionFactors)
+    addSaveWorkspace(m_outputFactorsWsName);
+}
+
+/**
+ * Handle mantid plotting
+ */
+void AbsorptionCorrections::plotClicked() {
+
+  QStringList plotData = {QString::fromStdString(m_pythonExportWsName),
+                          m_uiForm.dsSampleInput->getCurrentDataName()};
+  auto outputFactorsWsName =
+      m_absCorAlgo->getPropertyValue("CorrectionsWorkspace");
+  if (m_uiForm.ckKeepFactors->isChecked()) {
+    QStringList plotCorr = {QString::fromStdString(outputFactorsWsName) +
+                            "_ass"};
+    if (m_uiForm.ckUseCanCorrections->isChecked()) {
+      plotCorr.push_back(QString::fromStdString(outputFactorsWsName) + "_acc");
+      QString shiftedWs = QString::fromStdString(
+          m_absCorAlgo->getPropertyValue("CanWorkspace"));
+      plotData.push_back(shiftedWs);
     }
-    IndirectTab::plotSpectrum(plotData, 0);
+    plotSpectrum(plotCorr, 0);
   }
+  IndirectTab::plotSpectrum(plotData, 0);
```

Comment:
Is there a reason why sometimes we use `IndirectTab::method()` and sometimes just `method()`? For instance in this case, as far as I understand `plotSpectrum()` is defined and implemented in `IndirectTab` so there is no need to specify the base class here. It is OK if you want to explicitly say that `plotSpectrum` is implemented there, but if so I would do that everywhere for consistency. I've seen that with `checkADSForPlotSaveWorkspace()` too.